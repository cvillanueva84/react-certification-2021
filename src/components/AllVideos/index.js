export { default } from './AllVideos.component';