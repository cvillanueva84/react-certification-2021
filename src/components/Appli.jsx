import React from 'react';
import CardList from './CardList';
import NavBar from './NavBar';

export default function Appli() {
  return (
    <div>
      <NavBar />
      <CardList />
    </div>
  );
}
