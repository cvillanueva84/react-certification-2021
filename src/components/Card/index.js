export {default} from './Card.component.jsx'; 