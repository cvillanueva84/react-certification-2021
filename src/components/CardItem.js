import React from 'react';
import { useContext } from 'react';
import { GlobalContext } from './Context';
import {CardItemUi}  from './ui/CardItemUi'

export const CardItem = ({ title, description, url, videoId }) => {

  const { myStateReducer, dispatch } = useContext(GlobalContext);

  const handleClick = () => {
      dispatch({
        type: 'actionAddVideoData',
        payload: {
          ...myStateReducer,
          videoData: {
            title: title,
            description: description,
            url: url,
            videoId: videoId
          }
        }
      });
  }

  return (
    <CardItemUi 
      title={title} 
      description={description} 
      url={url} 
      handleClick={handleClick}
    />
  );

};

