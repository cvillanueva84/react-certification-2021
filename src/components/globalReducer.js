export const globalReducer = (state = {}, action) => {
    switch (action.type) {
        case 'actionLightMode':
            return {
                ...action.payload,
                mode: 'light',
            };

        case 'actionDarkMode':
            return {
                ...action.payload,
                mode: 'dark',
            };

        case 'actionAddSearch':
            return {
                ...action.payload,
            };

        case 'actionAddVideoData':
            return {
                ...action.payload,
            };

        case 'actionAddVideoDataFav':
            return {
                ...action.payload,
            };

        case 'actionAddFavorites':
            return {
                ...action.payload,
            };

        case 'actionDelete':
            return {
                ...state,
                favorites: state.favorites.filter((vid) => vid.videoId !== action.payload),
            };

        case 'actionLogin':
            return {
                ...action.payload,
                logged: true,
            };

        case 'actionLoginGoogle':
            return {
                ...action.payload,
                logged: true,
            };

        case 'actionLogout':
            return {
                ...action.payload,
                logged: false,
                user: '',
                videoDataFav: {},
            };

        default:
            return state;
    }
};
