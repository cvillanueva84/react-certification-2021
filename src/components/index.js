export { default as Navbar } from './Navbar/Navbar';
export { default as SearchBar } from './SearchBar/SearchBar';
export { default as ThemeSwitch } from './ThemeSwitch/ThemeSwitch';
export { default as Results } from './Results/Results';
