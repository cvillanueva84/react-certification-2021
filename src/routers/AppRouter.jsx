import React, { useContext, useEffect, useState } from 'react';
import { Switch, Redirect, Route } from 'react-router-dom';

import Favorites from '../components/Favorites';
import FavoriteVideoDetails from '../components/FavoriteVideoDetails';
import Home from '../components/Home/Home.component';
import Login from '../components/Login/Login.component';
import VideoDetails from '../components/VideoDetails';
import { AppContext } from '../context/appContext';
import { firebase } from '../firebase/firebase.config';
import { types } from '../types/types';
import { PrivateRoute } from './PrivateRoute';

export const AppRouter = () => {
  const { dispatch } = useContext(AppContext);
  const [isLoggedIn, setIsLoggedIn] = useState(false);

  useEffect(() => {
    firebase.auth().onAuthStateChanged(async (user) => {
      if (user?.uid) {
        dispatch({
          type: types.auth,
          payload: {
            isAuthenticated: true,
            auth: {
              uid: user.uid,
              email: user.email,
            },
          },
        });
        setIsLoggedIn(true);
      } else {
        setIsLoggedIn(false);
      }
    });
  }, [dispatch]);

  return (
    <Switch>
      <Route path="/" exact component={Home} />
      <Route
        path="/login"
        exact
        component={() => (isLoggedIn ? <Redirect to="/" /> : <Login />)}
      />
      <Route path="/video/:id" exact component={VideoDetails} />
      <PrivateRoute
        isAuthenticated={isLoggedIn}
        component={Favorites}
        path="/favorites"
        exact
      />
      <PrivateRoute
        isAuthenticated={isLoggedIn}
        component={FavoriteVideoDetails}
        path="/favorite/:id"
        exact
      />
      <Redirect to="/" />
    </Switch>
  );
};
