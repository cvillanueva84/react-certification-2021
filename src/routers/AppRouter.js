import React, { useContext } from 'react';
import { BrowserRouter as Router, Switch } from "react-router-dom";
import { GlobalContext } from '../components/Context';
import { LoginView } from '../components/LoginView';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

    const { myStateReducer } = useContext(GlobalContext);

    return (
        <Router>
            <>
                <Switch>
                    <PublicRoute 
                        exact path="/login" 
                        component={LoginView} 
                        isAuthenticated={myStateReducer.logged}
                    />

                    <PrivateRoute 
                        path="/" 
                        component={DashboardRoutes} 
                        isAuthenticated={myStateReducer.logged}
                    />
                </Switch>
            </>
        </Router>
    )
}
