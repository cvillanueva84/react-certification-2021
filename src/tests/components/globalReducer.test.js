import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router';
import { GlobalContext } from '../../components/Context';
import { AppRouter } from "../../routers/AppRouter";
import { globalReducer } from '../../components/globalReducer';

describe('Test globalReducer', () => {
    const contextValue = {
        dispatch: jest.fn(),
        myStateReducer: {
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }
    }

    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const wrapper = mount(
        <GlobalContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <AppRouter />
                </Router>
            </MemoryRouter>
        </GlobalContext.Provider>
    );

    afterEach(() => {
        jest.clearAllMocks();
    })

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.exists());
    });


    it('should initialState', () => {
        const state = globalReducer({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }, {})
        expect(state).toEqual({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        })
    });

    it('should initialState', () => {
        const state = globalReducer({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }, {})
        expect(state).toEqual({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        })
    });

    it('should actionLightMode', () => {
        const action = {
            type: 'actionLightMode',
            payload: {
                mode: 'light'
            }
        };

        const state = globalReducer({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }, action);

        expect(state).toEqual({
            mode: 'light',
        })
    });

    it('should actionDarkMode', () => {
        const action = {
            type: 'actionDarkMode',
            payload: {
                mode: 'dark'
            }
        };

        const state = globalReducer({
            mode: 'dark',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }, action);

        expect(state).toEqual({
            mode: 'dark',
        })
    });

    it('should actionAddSearch', () => {
        const action = {
            type: 'actionAddSearch',
            payload: {
                mode: 'light',
                search: 'wizeline',
                videoData: {},
                videoDataFav: {},
                user: '',
                logged: false,
                favorites: [],
            }
        };

        const state = globalReducer({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }, action);

        expect(state).toEqual({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        })
    });

    it('should actionAddVideoData', () => {
        const action = {
            type: 'actionAddVideoData',
            payload: {
                mode: 'light',
                search: 'wizeline',
                videoData: {},
                videoDataFav: {},
                user: '',
                logged: false,
                favorites: [],
            }
        };

        const state = globalReducer({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }, action);

        expect(state).toEqual({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        })
    });

    it('should actionAddVideoDataFav', () => {
        const action = {
            type: 'actionAddVideoDataFav',
            payload: {
                mode: 'light',
                search: 'wizeline',
                videoData: {},
                videoDataFav: {},
                user: '',
                logged: false,
                favorites: [],
            }
        };

        const state = globalReducer({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }, action);

        expect(state).toEqual({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        })
    });

    it('should actionAddFavorites', () => {
        const action = {
            type: 'actionAddFavorites',
            payload: {
                mode: 'light',
                search: 'wizeline',
                videoData: {},
                videoDataFav: {},
                user: '',
                logged: false,
                favorites: [],
            }
        };

        const state = globalReducer({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }, action);

        expect(state).toEqual({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        })
    });

    it('should actionDelete', () => {
        const action = {
            type: 'actionDelete',
            payload: {
                mode: 'light',
                search: 'wizeline',
                videoData: {},
                videoDataFav: {},
                user: '',
                logged: false,
                favorites: [],
            }
        };

        const state = globalReducer({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }, action);

        expect(state).toEqual({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        })
    });

    it('should actionLogin', () => {
        const action = {
            type: 'actionLogin',
            payload: {
                logged: true
            }
        };

        const state = globalReducer({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }, action);

        expect(state).toEqual({
            logged: true,
        })
    });

    it('should actionLoginGoogle', () => {
        const action = {
            type: 'actionLoginGoogle',
            payload: {
                logged: true
            }
        };

        const state = globalReducer({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }, action);

        expect(state).toEqual({
            logged: true,
        })
    });

    it('should actionLogout', () => {
        const action = {
            type: 'actionLogout',
            payload: {
                logged: false,
                user: '',
                videoDataFav: {},
            }
        };

        const state = globalReducer({
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }, action);

        expect(state).toEqual({
            logged: false,
            user: '',
            videoDataFav: {},
        })
    });


});