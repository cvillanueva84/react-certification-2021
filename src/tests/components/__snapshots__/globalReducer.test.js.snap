// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test globalReducer should render 1`] = `ReactWrapper {}`;
